abm/connector.py

```diff
@@ -313,3 +313,21 @@ def write_dataset(self, fptr, length):
         tmp_catalog.close()
         # TODO: Need to figure out how to handle error return
         return True
+
+    def write_dataset_bytes(self, bytes):
```

Comment:
Hi. I see that write_dataset_bytes() is similar to write_dataset().
Let's discuss the difference.
Is there a way for one of these methods to call the other, to reduce the code replication?

Reply:
Each function gets a different parameters, but maybe we can reduce the code duplication.